src/test/java/org/dcsa/api_validator/ovs/v2/PostTimestampsTest.java

```diff
@@ -0,0 +1,588 @@
+                then().
+                assertThat().
+                statusCode(400);
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                header("testname","testTransportCallsNoMandatoryParameters2").
+                body("").
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+    }
+
+    // These are mandatory parameters and is the main test
+    @Test
+    public void testTimestampRequiredParameters() {
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                header("testname","testTimestampRequiredParameters").
+                body(VALID_TIMESTAMP).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(204);
+    }
+
+    @Test
+    public void testFacilitySMDGCodeQueryParam() {
+
+        Map<String, String> map = (Map<String, String>) jsonToMap(VALID_TIMESTAMP);
+        map.remove("eventLocation");
+        map.remove("vesselPosition");
+        map.remove("modeOfTransport");
+        map.remove("portCallServiceTypeCode");
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                header("testname","testFacilitySMDGCodeQueryParam").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(204);
+    }
+
+    // Test facilitySMDGCode, fails as specification -> maxLength:6 & ENUM
+    @Test
+    public void testFacilitySMDGCodeFalseFormat() {
+
+        Map<String, String> map = (Map<String, String>) jsonToMap(VALID_TIMESTAMP);
+        map.remove("eventLocation");
+        map.remove("vesselPosition");
+        map.remove("modeOfTransport");
+        map.remove("portCallServiceTypeCode");
+        map.put("facilitySMDGCode","aBCDaFGHaE");
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+
+        map.put("facilitySMDGCode", "aBC");
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+    }
+
+    @Test
+    public void testEventLocationQueryParam() {
```

Comment:
I think this test name is wrong - as it is not testing query parameters. It is testing eventLocation field in a timestamp